### DISCUSSION TITLE
Prebuild, config plugins and cocoapods config

### DISCUSSION THREAD

**Discussion by lHersey** [expo/expo#13417] [Archived: Questions] [answered]
Hi, I'm trying to learn how to use everything that is new on expo, I saw the documentation and the blog but found no answer.

I'm trying to use a RN package that have native modules, so as far as I understand, I need a config plugin that makes the config for me at prebuild time (Add required keys on androidmanifest or infoPlist for example),  The package does not have a config plugin, I wanted to create it by myself. Since we can generate the ios/android with `prebuild/run` 

I don't think is necessary to commit those folders since I'm not going to modify them directly (only via config plugins) and I can generate them at any time with prebuild or run commands

But I didn't see any info about how to modify the generated cocapods files for RN packages that requires to add a new line to that file.

The package in question: https://github.com/react-native-text-input-mask/react-native-text-input-mask (I chosed this package for learning purposes, I don't need it in a real project)

The documentation has these mods:
mods.ios.appDelegate -- Modify the ios/<name>/AppDelegate.m as a string.
mods.ios.infoPlist -- Modify the ios/<name>/Info.plist as JSON (parsed with @expo/plist).
mods.ios.entitlements -- Modify the ios/<name>/<product-name>.entitlements as JSON (parsed with @expo/plist).
mods.ios.expoPlist -- Modify the ios/<name>/Expo.plist as JSON (Expo updates config for iOS) (parsed with @expo/plist).
mods.ios.xcodeproj -- Modify the ios/<name>.xcodeproj as an XcodeProject object (parsed with xcode).
mods.android.manifest -- Modify the android/app/src/main/AndroidManifest.xml as JSON (parsed with xml2js).
mods.android.strings -- Modify the android/app/src/main/res/values/strings.xml as JSON (parsed with xml2js).
mods.android.colors -- Modify the android/app/src/main/res/values/colors.xml as JSON (parsed with xml2js).
mods.android.styles -- Modify the android/app/src/main/res/values/styles.xml as JSON (parsed with xml2js).
mods.android.mainActivity -- Modify the android/app/src/main/<package>/MainActivity.java as a string.
mods.android.appBuildGradle -- Modify the android/app/build.gradle as a string.
mods.android.projectBuildGradle -- Modify the android/build.gradle as a string.
mods.android.settingsGradle -- Modify the android/settings.gradle as a string.
mods.android.gradleProperties -- Modify the android/gradle.properties as a Properties.PropertiesItem[].

Any help or guide is welcome, thanks!

**Comment by brentvatne** [comment, marked as answer]
you don't need a config plugin for this package, the native dependencies will be automatically linked and this library doesn't require any additional configuration.

edit: nevermind! i see that there is some additional configuration needed for RCTText... in this case, you need to modify the Podfile as a string basically, you can see an example of how this is done in the [react-native-maps plugin built into expo-cli](https://github.com/expo/expo-cli/blob/master/packages/config-plugins/src/ios/Maps.ts). ideally all of this would be handled with autolinking so manual modification of these files isn't needed, we're working towards that

**Reply by lHersey** [reply]
But in the docs I found this.

![image](https://user-images.githubusercontent.com/18200355/123531454-0de45780-d6c2-11eb-861a-13144ee8f474.png)

And the generated cocoapods file (prebuild) does not have the `use_frameworks!` statement.

Can you explain why I don't need to configure anything? 

Thanks for the reply

**Reply by lHersey** [reply]
Awesome! Will check the example, thanks! :D

**Reply by EvanBacon** [reply]
Be sure to document any regex based plugins as “unstable” or “dangerous” since they depend on the state of the template to be a certain way (which might not age well in a few years). Like Brent said, we’ll build all of these features into Expo autolinking and Expo modules so plugins can safely interact with a code base.

**Reply by JaapWeijland** [reply]
But as I understand from this thread, if I want to add a package that only needs to be autolinked, I just "expo install <package>" and the package gets included in my prebuild? So plugins are only necessary for packages that need for example Info.plist modified?
